alibi_detect/cd/base_online.py

```diff
@@ -106,6 +113,66 @@ def _configure_ref_subset(self):
     def _update_state(self, x_t: Union[np.ndarray, 'tf.Tensor', 'torch.Tensor']):
         pass
 
+    def _set_state_dir(self, dirpath: Union[str, os.PathLike]):
+        """
+        Set the directory path to store state in, and create an empty directory if it doesn't already exist.
+
+        Parameters
+        ----------
+        dirpath
+            The directory to save state file inside.
+        """
+        self.state_dir = Path(dirpath)
```

Comment:
Should this be a private attribute?

Reply:
See https://github.com/SeldonIO/alibi-detect/pull/604#discussion_r1069655715